book/src/libs/wasp/benchspy/prometheus_std.md

```diff
@@ -0,0 +1,112 @@
+# BenchSpy - Standard Prometheus Metrics
+
+Now that we've seen how to query and assert load-related metrics, let's explore how to query and assert on resource usage by our Application Under Test (AUT).
+
+If you're unsure why this is important, consider the following situation: the p95 latency of a new release matches the previous version, but memory consumption is 34% higher. Not ideal, right?
+
+## Step 1: Prometheus Configuration
+
+Since `WASP` has no built-in integration with `Prometheus`, we need to pass its configuration separately:
+
+```go
+promConfig := benchspy.NewPrometheusConfig("node[^0]")
+```
+
+This constructor loads the URL from the environment variable `PROMETHEUS_URL` and adds a single regex pattern to match containers **by name**. In this case, it excludes the bootstrap Chainlink node (named `node0` in the `CTFv2` stack).
+
+> [!WARNING]
+> This example assumes that you have both the observability stack and basic node set running.
+> If you have the `CTF CLI`, you can start it by running: `ctf b ns`.
+
+> [!NOTE]
+> Matching containers **by name** should work both for most k8s and Docker setups using `CTFv2` observability stack.
+
+## Step 2: Fetching and Storing a Baseline Report
+
+As in previous examples, we'll use built-in Prometheus metrics and fetch and store a baseline report:
+
+```go
+baseLineReport, err := benchspy.NewStandardReport(
+    "91ee9e3c903d52de12f3d0c1a07ac3c2a6d141fb",
+    // notice the different standard query executor type
+    benchspy.WithStandardQueries(benchspy.StandardQueryExecutor_Prometheus),
+    benchspy.WithPrometheusConfig(promConfig),
+    // Required to calculate test time range based on generator start/end times.
+    benchspy.WithGenerators(gen),
+)
+require.NoError(t, err, "failed to create baseline report")
+
+fetchCtx, cancelFn := context.WithTimeout(context.Background(), 60*time.Second)
+defer cancelFn()
+
+fetchErr := baseLineReport.FetchData(fetchCtx)
+require.NoError(t, fetchErr, "failed to fetch baseline report")
+
+path, storeErr := baseLineReport.Store()
+require.NoError(t, storeErr, "failed to store baseline report", path)
+```
+
+> [!NOTE]
+> Standard metrics for Prometheus differ from those used by `Loki` or `Direct` query executors.
+> Prometheus metrics focus on resource usage by the AUT, while `Loki`/`Direct` metrics measure load characteristics.
+>
+> Standard Prometheus metrics include:
+> - `median_cpu_usage`
+> - `median_mem_usage`
+> - `p95_cpu_usage`
+> - `p95_mem_usage`
```

Comment:
Do we care about "disk space" regressions, .e.g. new version started writing 10x logs so by end of tests exhausted space. If yes, may be worth adding

Reply:
We are not certain if the raw log amount matters. If we assert it in any report, another run with debug logs enabled cannot be compared. Instead, it might be better to implement SQL benchspy later and assert total table size and per-table size for CL nodes, if applicable. Most node data is disposable, except for jobs, which don’t take up much space. Still, I believe tracking the total table size will be useful.
We are also not I/O heavy on the database size but it's worth asserting the network activity of nodes/services, I believe we can do that in the next PRs.